Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -163,117 +205,34 @@ public void Serialize<T>(IIonWriter writer, T item)
         {
             if (item == null)
             {
-                new IonNullSerializer().Serialize(writer, (object)null);
+                this.nullSerializer.Serialize(writer, null);
                 return;
             }
 
-            if (item is bool)
+            Type type = item.GetType();
+            if (this.primitiveSerializers.ContainsKey(type))
             {
-                new IonBooleanSerializer().Serialize(writer, Convert.ToBoolean(item));
+                this.SerializePrimitive(type, writer, item);
                 return;
             }
 
-            if (item is int)
+            if (item is IList)

```

Comment:
New C# compiler supports type in switch, can we use it here.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/switch

Reply:
Ian seems to dislike switch statements.